docs/import/color_function_util.js

```diff
@@ -25,22 +191,149 @@ function withColor(td, layer, property, index) {
       td.append(createColorBox(colorFunction['color']));
       break;
     case ColorStyle.DISCRETE:
-      const colorObject = colorFunction['colors'];
-      const colorSet = new Set();
-      Object.keys(colorObject).forEach((propertyValue) => {
-        const color = colorObject[propertyValue];
-        if (!colorSet.has(color)) {
-          colorSet.add(color);
-          td.append(createColorBox(colorObject[propertyValue]));
-        }
-      });
+      createColorBoxesForDiscrete(colorFunction, td);
       break;
     default:
       setStatus(ILLEGAL_STATE_ERR + 'unrecognized color function: ' + layer);
   }
+  td.addClass('editable')
+      .on('click', () => onClick(td, colorFunction['current-style']));
   return td;
 }
 
+/**
+ * On click function for color tds - updates the globalTd.
+ * @param {JQuery<HTMLElement>} td
+ * @param {enum} type
+ */
+function onClick(td, type) {
+  if ($(td).hasClass('na')) {
+    return;
+  }
+  const colorFunctionDiv = $('#color-fxn-editor');
+  if (colorFunctionDiv.is(':visible') && td === globalTd) {
+    colorFunctionDiv.hide();
+    $(globalTd).removeClass('selected');
+    return;
+  }
+  colorFunctionDiv.show();
+  if (td === globalTd) {
+    $(globalTd).addClass('selected');
+    return;
+  }
+  $(globalTd).removeClass('selected');
+  globalTd = td;
+  $(globalTd).addClass('selected');
+  $('#' + colorStyleTypeStrings.get(type) + '-radio')
```

Comment:
So we're assuming there is always a default type already set? Maybe we can make that SINGLE_COLOR on layer creation?

Reply:
Yep, that's what I was thinking. Maybe they also pick a starting color but I guess that doesn't need to happen.